ouroboros-consensus-byron-test/src/Test/Consensus/Byron/Generators.hs

```diff
@@ -77,6 +86,82 @@ epochSlots = EpochSlots 100
 protocolMagicId :: ProtocolMagicId
 protocolMagicId = ProtocolMagicId 100
 
+instance Arbitrary CC.Genesis.Config where
+  arbitrary = CC.Genesis.Config
+    <$> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+
+instance Arbitrary CC.Genesis.GenesisData where
+  arbitrary = CC.Genesis.GenesisData
+    <$> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+
+instance Arbitrary CC.Genesis.GenesisKeyHashes where
+  arbitrary = CC.Genesis.GenesisKeyHashes <$> arbitrary
+
+instance Arbitrary CC.Genesis.GenesisDelegation where
+  arbitrary = (CC.Genesis.mkGenesisDelegation <$> arbitrary)
+    `suchThatMap` (either (const Nothing) Just)
+
+instance Arbitrary (CC.Del.ACertificate ()) where
+  arbitrary = CC.Del.signCertificate
+    <$> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+
+instance Arbitrary SafeSigner where
```

Comment:
There is a `genSafeSigner` generator in `cardano-ledger-specs/byron/crypto/test/Test/Cardano/Crypto/Gen.hs`, but it leaves the passphrase blank rather than using an arbitrary pass phrase. According to @dcoutts, the passphrase "was security theatre. It added no value and considerably complexity". That said, we probably still want to generate some passphrase for the serialization tests, in which case, this instance is perhaps appropriate.